### PR TITLE
fix: AccessEntry API representation parsing

### LINKED ISSUE #1681: `AccessEntry` incorrectly parsed from API Representation
Thanks for stopping by to let us know something could be better!

**PLEASE READ**: If you have a support contract with Google, please create an issue in the [support console](https://cloud.google.com/support/) instead of filing on GitHub. This will ensure a timely response.

Please run down the following list and make sure you've tried the usual "quick fixes":

  - Search the issues already opened: https://github.com/googleapis/python-bigquery/issues
  - Search StackOverflow: https://stackoverflow.com/questions/tagged/google-cloud-platform+python

If you are still having issues, please be sure to include as much information as possible:

#### Environment details

  - OS type and version: Debian GNU/Linux 11 (bullseye)
  - Python version: Python 3.10.13
  - pip version: pip 23.2.1
  - `google-cloud-bigquery` version: 3.11.4

#### Steps to reproduce

  1. Create an `AccessEntry` through `AccessEntry.from_api_repr` without a role in the payload.
    a. This is how the BigQuery API returns `view` type `AccessEntry`s.
  2. Create the same `AccessEntry` through `AccessEntry.__init__` with the equivalent resource.
  3. Compare the 2 for equality and get `False`

#### Code example

```python
from google.cloud.bigquery.dataset import AccessEntry

resource = {
    'view': {
        'projectId': 'my-project',
        'datasetId': 'my_dataset',
        'tableId': 'my_table',
    }
}

api_repr_entry = AccessEntry.from_api_repr(resource)
expected_entry = AccessEntry(
    entity_type='view',
    entity_id=resource['view'],
)

print(api_repr_entry == expected_entry) # False

>>> print(api_repr_entry._properties) # {'view': {'projectId': 'my-project', 'datasetId': 'my_dataset', 'tableId': 'my_table'}}
>>> print(expected_entry._properties) # {'view': {'projectId': 'my-project', 'datasetId': 'my_dataset', 'tableId': 'my_table'}, 'role': None}
```

The entry created through `AccessEntry.__init__` has `'role': None` in its `AccessEntry#_properties` while the one from `AccessEntry.from_api_repr` does not causing the equality to fail. This is due to the `_properties` being overwritten with the API resource which doesn't include the `role` key when it's `null`.

Making sure to follow these steps will guarantee the quickest resolution possible.

Thanks!

### PR DESCRIPTION
**Pull request by jonathan-ostrander**
Overriding the `AccessEntry#_properties` with a deep copy of the API resource overwrites the `role` property set in `AccessEntry.__init__` which isn't present in the resource if the `role` is set to `None`. This causes `AccessEntry`s generated from API representations to no longer evaluate to equal with equivalent `AccessEntry` resources instantiated through `AccessEntry.__init__`. The added unit test fails without the change and passes with the change.

Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [x] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigquery/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [x] Ensure the tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

Fixes #1681 🦕

### CODE REVIEW

**Comment by Linchin** [comment]
Hi @jonathan-ostrander, thank you so much for your contribution! The PR looks good, but it seems the lint test failed. As seen in the kokoro results:
image

Please see the [instructions in the contribution doc](https://github.com/googleapis/python-bigquery/blob/main/CONTRIBUTING.rst#coding-style) for more info on coding style. If you need any help, please let us know! :)

File: tests/unit/test_dataset.py [hunk 1]
```diff
@@ -152,6 +152,22 @@ def test_from_api_repr_w_unknown_entity_type(self):
         exp_resource = entry.to_api_repr()
         self.assertEqual(resource, exp_resource)
```
Comment (Linchin) [inline tests/unit/test_dataset.py:154]: please fix linting test

**Review by Linchin — CHANGES_REQUESTED** [review]
please fix linting error

**Comment by jonathan-ostrander** [comment]
@Linchin should be fixed in [15153e6](https://github.com/googleapis/python-bigquery/pull/1682/commits/15153e6980e692a3c3a87085fd9be6a32ecea4b3) although I couldn't get nox working locally so I just ran black.

**Review by Linchin — APPROVED** [review]

**Pull request merged**